### PR TITLE
HOSTEDCP-1094: e2e autoscaler balancing similar node groups

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
Updates autoscaler e2e to validate `--balance-similar-node-groups` feature of the autoscaler 

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-1094](https://issues.redhat.com/browse/HOSTEDCP-1094)

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by enxebre** [comment]
very nice.
/approve
/test e2e-aws

**Comment by enxebre** [comment]
very nice!
/approve
/test e2e-aws
To check flakes in case scale down is not deterministic

File: test/e2e/autoscaling_test.go [hunk 1]
```diff
@@ -30,68 +31,77 @@ func TestAutoscaling(t *testing.T) {
...
+		err := e2eutil.UpdateObject(t, ctx, client, &nodepool, mutateFunc)
+		g.Expect(err).ToNot(HaveOccurred(), "failed to update NodePool %s", nodepool.Name)
 	}
-	nodepool.Spec.Replicas = nil
-	err = client.Patch(ctx, nodepool, crclient.MergeFrom(original))
-	g.Expect(err).NotTo(HaveOccurred(), "failed to update NodePool")
-	t.Logf("Enabled autoscaling. Namespace: %s, name: %s, min: %v, max: %v", nodepool.Namespace, nodepool.Name, min, max)
+	t.Logf("Enabled autoscaling for all nodePools in namespace: %s, min: %v, max: %v", hostedCluster.Namespace, min, max)
 
 	// TODO (alberto): check autoscalingEnabled condition.
```
[the remark this answers is not in the document]
Comment (muraee) [reply test/e2e/autoscaling_test.go:70]: yeah I guess we can

**Comment by muraee** [comment]
`TestCreateCluster/ValidateMetricsAreExposed` failed for some reason.
/retest-required

**Review by bryan-cox — APPROVED** [review]
/lgtm

File: test/e2e/autoscaling_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (enxebre) [reply test/e2e/autoscaling_test.go]: I don't see where this changes cover this todo, where are we checking the autoscaling condition is as expected now?

**Pull request merged**